Library/Homebrew/extend/ARGV.rb

```diff
@@ -37,6 +44,10 @@ def kegs
     @kegs ||= downcased_unique_named.collect do |name|
       canonical_name = Formulary.canonical_name(name)
       rack = HOMEBREW_CELLAR/canonical_name
+
+      # Cellar/oldname can be a symlink to Cellar/newname, so in order not to # get newname's keg as oldname's keg we check if it's a symlink.
+      raise "#{rack.to_s} is a symlink" if rack.symlink?
```

Comment:
This happens on uninstall when we're trying to uninstall something that is a symlink.